src/components/sporsmal/tilleggssporsmal/Egenmeldingsdager.tsx

```diff
@@ -1,89 +1,75 @@
 import React, { useState, useEffect } from 'react';
-import { FieldError, ValidationPayload } from 'react-hook-form/dist/types';
-import Periodevelger from '../periodevelger/Periodevelger';
-import { Knapp } from 'nav-frontend-knapper';
-import { SkjemaGruppe, Fieldset } from 'nav-frontend-skjema';
+import { useFormContext } from 'react-hook-form';
 import tekster from '../sporsmal-tekster';
+import { SkjemaGruppe, Fieldset } from 'nav-frontend-skjema';
+import Lenke from 'nav-frontend-lenker';
+import { Egenmeldingsperiode } from '../valideringsSkjema';
+import Flatpickr from 'react-flatpickr';
+import './flatpickr.less';
+import './egenmeldingsdager.less';
+import { locale } from '../../../types/sporsmalTypes';
 
 interface EgenmeldingsdagerProps {
-    vis: boolean;
+    name: string;
     sykmeldingStartdato?: Date;
-    register: any;
-    unregister: any;
-    triggerValidation: (
-        payload?: ValidationPayload<string, any> | ValidationPayload<string, any>[] | undefined,
-        shouldRender?: any,
-    ) => Promise<boolean>;
-    isSubmitted: boolean;
-    setValue: (name: string, startOgSlutt: any, shouldValidate?: boolean) => void;
-    errors: Partial<Record<string, FieldError>>;
 }
 
-export interface Egenmeldingsperiode {
-    id: number;
-    startDato: Date | null;
-    sluttDato: Date | null;
-}
+const Egenmeldingsdager = ({ name, sykmeldingStartdato }: EgenmeldingsdagerProps) => {
+    const { register, unregister, errors, setValue, triggerValidation, formState } = useFormContext();
+    const { isSubmitted } = formState;
 
-const Egenmeldingsdager = ({
-    vis,
-    sykmeldingStartdato,
-    register,
-    unregister,
-    setValue,
-    errors,
-    triggerValidation,
-    isSubmitted,
-}: EgenmeldingsdagerProps) => {
-    const [perioder, setPerioder] = useState<Egenmeldingsperiode[]>([{ id: 0, startDato: null, sluttDato: null }]); // Legger til første periode
-    const name = 'egenmeldingsperioder';
+    const [perioder, setPerioder] = useState<Egenmeldingsperiode[]>([{ id: 0 }]); // Legger til første periode
 
     // Registrer ved mount, unregistrer ved unmount
     useEffect(() => {
-        register({ name: name });
-        return () => unregister(name);
-    }, [register, unregister]);
+        register({ name });
+        return () => {
+            setValue(name, undefined);
+            unregister(name);
+        };
+    }, [name, register, setValue, unregister]);
+
+    const opprettNyPeriode = (): void => {
+        const nyPeriode: Egenmeldingsperiode = { id: perioder[perioder.length - 1].id + 1 };
```

Comment:
Finnes det tilfeller der denne måten å sette ID på vil føre til duplikater (som da kan påvirke sletting)?

Mulig det er tryggere å bruke et bibliotek som feks 'uuid' for å generere unike IDer.

Reply:
Tror det skal være sikret mot å ha samme id, dog mer ryddig med uuid. skal se på det :)

Reply:
tenkte litt mer på denne. kanskje unngå å dra inn flere biblioteker enn vi må? 

Slik id settes nå, så vil den alltid plusse på én fra id-en til elementet som er sist i listen, som allerede vil ha den høyeste id-en i listen.
Hvis man legger til to perioder vil de har id: `0, 1, 2`. Sletter man periode nr 2 vil id være `0, 2`. Legger man til en tredje periode igjen, vil id være `0, 2, 3`.

Reply:
Enig, fant ikke selv en måte å ødelegge på 💯